docs(lsp): simplify example of enabling LSP folding (#32896)
neovim/neovim: runtime/doc/lsp.txt

    Provides an interface between the built-in client and a `foldexpr`
    function.

    To use, set 'foldmethod' to "expr" and set the value of 'foldexpr': >lua
        vim.o.foldmethod = 'expr'
        vim.o.foldexpr = 'v:lua.vim.lsp.foldexpr()'
<

>     Or use it on demand by checking for the "textDocument/foldingRange"

Reviewer: ```suggestion
    Or use it only when supported by checking for the "textDocument/foldingRange"
```

    capability in an |LspAttach| autocommand. Example: >lua

>         -- General folding preferences

Reviewer: ```suggestion
```

    Provides an interface between the built-in client and a `foldexpr`
    function.

>     To use, set 'foldmethod' to "expr" and set the value of 'foldexpr': >lua

Reviewer 2: I would actually just include this part as it's unclear when/why to use the "on demand" approach from this example.

Author: Sounds reasonable, I force-pushed an update. Changed this to an example using both tree-sitter and LSP.

Additionally, I did not choose to delete this part directly because this example may help users learn some knowledge about how to use the LSP API.
